source/GameInterface/Services/Heroes/Patches/TakePrisonerPatch.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Common;
+using Common.Extensions;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Policies;
+using GameInterface.Services.Heroes.Messages;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+
+namespace GameInterface.Services.Heroes.Patches
+{
+    [HarmonyPatch(typeof(TakePrisonerAction))]
+    public class TakePrisonerPatch
+    {
+        private static readonly Action<PartyBase, Hero, bool> ApplyInternal =
+            typeof(TakePrisonerAction)
+            .GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static)
+            .BuildDelegate<Action<PartyBase, Hero, bool>>();
+
+        [HarmonyPrefix]
+        [HarmonyPatch("ApplyInternal")]
+        private static bool Prefix(PartyBase capturerParty, Hero prisonerCharacter, bool isEventCalled = true)
+        {
+            if (AllowedThread.IsThisThreadAllowed()) return true;
+
+            if (PolicyProvider.AllowOriginalCalls) return true;
+
+            if (ModInformation.IsClient && prisonerCharacter != Hero.MainHero) return false;
```

Comment:
Server should be able to control the release here

Reply:
Not sure if server can fully control this as it gets called from UI, I'll look into it.